### PR TITLE
Feature/7252 get catalogue order item

### PR DESCRIPTION
**Pull request by RuthBerry96**
Story: [AB#7252](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7252)
Task: [AB#7863](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7863)

### CODE REVIEW

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/EntityBuilder/OrderItemEntityBuilder.cs [hunk 1]
```diff
@@ -21,6 +21,8 @@ public sealed class OrderItemEntityBuilder
         private string _pricingUnitTierName;
         private string _pricingUnitDescription;
         private decimal? _price;
+        private DateTime _created;
+        private DateTime _lastUpdated;
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/EntityBuilder/OrderItemEntityBuilder.cs:25]: Can these be `readonly`?

File: src/NHSD.BuyingCatalogue.Ordering.Api/Models/CatalogueSolutionModel.cs [hunk 2]
```diff
@@ -2,6 +2,8 @@
 {
     public sealed class CatalogueSolutionModel
     {
-        public string Name { get; set; }
+        public int OrderItemId { get; set; }
+        public string SolutionName { get; set; }
+        public GetServiceRecipientModel ServiceRecipient { get; set; }
     }
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.Api/Models/CatalogueSolutionModel.cs:8]: 👍

File: src/NHSD.BuyingCatalogue.Ordering.Domain/Order.cs [hunk 3]
```diff
@@ -77,5 +82,24 @@ public void SetLastUpdatedBy(Guid userId, string name)
 
             SetLastUpdatedBy(userId, name);
         }
+
+        public void SetServiceRecipient(List<(string Ods, string Name)> serviceRecipients, Guid userId, string name)
+        {
+            if (serviceRecipients is null)
+                throw new ArgumentNullException(nameof(serviceRecipients));
+
+            _serviceRecipients.Clear();
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.Domain/Order.cs:91]: no, you're right. I normally put a blank line in, just obviously not this time!

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Catalogue Solutions Get All.feature [hunk 4]
```diff
@@ -7,40 +7,64 @@ Background:
     Given Orders exist
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Catalogue Solutions Get All.feature:7]: 😱

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Catalogue Solutions Get All.feature [hunk 5]
```diff
@@ -7,40 +7,64 @@ Background:
...
         | OrderId    | Description         | OrderStatusId | LastUpdatedBy                        | OrganisationId                       |
         | C000014-01 | Some Description    | 1             | 335392e4-4bb1-413b-9de5-36a85c9c0422 | 4af62b99-638c-4247-875e-965239cd0c48 |
+        | C000014-02 | Another Description | 1             | 335392e4-4bb1-413b-9de5-36a85c9c0422 | 4af62b99-638c-4247-875e-965239cd0c48 |
+    Given Service Recipients exist
+        | OrderId    | OdsCode | Name     |
+        | C000014-01 | eu      | EU Test  |
+        | C000014-01 | auz     | AUZ Test |
+        | C000014-01 | ods     | NULL     |
+    Given OrderItem exist
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Catalogue Solutions Get All.feature:16]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CatalogueSolutionsControllerTests.cs [hunk 6]
```diff
@@ -48,18 +51,88 @@ public async Task UpdateAsync_ForExistingOrder_UpdatesCatalogueSolutionsViewed()
         }
 
         [Test]
-        public async Task GetAllAsync_WithDescription_ReturnsOkResult()
+        public async Task GetAllAsync_WithDescriptionAndNoSolution_ReturnsOkResult()
         {
             var expectedDescription = "A description";
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CatalogueSolutionsControllerTests.cs:56]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CatalogueSolutionsControllerTests.cs [hunk 7]
```diff
@@ -48,18 +51,88 @@ public async Task UpdateAsync_ForExistingOrder_UpdatesCatalogueSolutionsViewed()
...
             result.Value.Should().BeOfType<CatalogueSolutionsModel>();
             var model = result.Value;
             model.CatalogueSolutions.Should().BeEmpty();
             model.OrderDescription.Should().Be(expectedDescription);
         }
 
+        [Test]
+        public async Task GetAllAsync_WithSolution_ReturnsOkayResult()
+        {
+            var expectedDescription = "A description";
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CatalogueSolutionsControllerTests.cs:70]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CatalogueSolutionsControllerTests.cs [hunk 8]
```diff
@@ -48,18 +51,88 @@ public async Task UpdateAsync_ForExistingOrder_UpdatesCatalogueSolutionsViewed()
...
+
+            var context = CatalogueSolutionsControllerTestContext.Setup();
+
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CatalogueSolutionsControllerTests.cs:73]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CatalogueSolutionsControllerTests.cs [hunk 9]
```diff
@@ -48,18 +51,88 @@ public async Task UpdateAsync_ForExistingOrder_UpdatesCatalogueSolutionsViewed()
...
+            model.OrderDescription.Should().Be(expectedDescription);
+            model.CatalogueSolutions.Should().BeEquivalentTo(expectedCatalogueSolutionList);
+        }
+
+        [Test]
+        public async Task GetAllAsync_MultipleSolutions_ReturnsOkayResult()
+        {
+            var context = CatalogueSolutionsControllerTestContext.Setup();
+
+            var expectedDescription = "A description";
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CatalogueSolutionsControllerTests.cs:105]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Domain.UnitTests/Builders/OrderItemBuilder.cs [hunk 10]
```diff
@@ -3,7 +3,7 @@
 
 namespace NHSD.BuyingCatalogue.Ordering.Domain.UnitTests.Builders
 {
-    internal sealed class OrderItemBuilder
+    public sealed class OrderItemBuilder
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Domain.UnitTests/Builders/OrderItemBuilder.cs:6]: 👍

File: tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/EntityBuilder/OrderItemEntityBuilder.cs (hunk 1 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/EntityBuilder/OrderItemEntityBuilder.cs:25]: 👍

File: src/NHSD.BuyingCatalogue.Ordering.Domain/Order.cs [hunk 11]
```diff
@@ -77,5 +82,24 @@ public void SetLastUpdatedBy(Guid userId, string name)
...
+            foreach (var recipient in serviceRecipients)
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.Domain/Order.cs:92]: It is clearer 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CatalogueSolutionsControllerTests.cs [hunk 12]
```diff
@@ -319,5 +392,18 @@ internal static CatalogueSolutionsControllerTestContext Setup()
                 return new CatalogueSolutionsControllerTestContext();
             }
         }
+        private CatalogueSolutionModel CreateCatalogueSolutionModel(OrderItem orderItem, (string Ods, string Name) serviceRecipient)
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CatalogueSolutionsControllerTests.cs:395]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CatalogueSolutionsControllerTests.cs [hunk 13]
```diff
@@ -48,18 +51,87 @@ public async Task UpdateAsync_ForExistingOrder_UpdatesCatalogueSolutionsViewed()
...
+            model.OrderDescription.Should().Be(expectedDescription);
+            model.CatalogueSolutions.Should().BeEquivalentTo(expectedCatalogueSolutionList);
+        }
+
+        [Test]
+        public async Task GetAllAsync_MultipleSolutions_ReturnsOkayResult()
+        {
+            var context = CatalogueSolutionsControllerTestContext.Setup();
+
+            const string  expectedDescription = "A description";
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CatalogueSolutionsControllerTests.cs:104]: There's a double space here.

Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CatalogueSolutionsControllerTests.cs:104]: 😱

**Pull request merged**